8.0 sms

Viggor opened the pull request:
Add modules to send sms to partner with ovh configuration.
Add a module to automatic sms sending when a deliver is ready.

The original sms module come from Julius.

yvanpatry committed:
[ADD] Add module SMS client , still some bug to fix

yvanpatry committed:
[IMP] Udapte Module Information

yvanpatry committed:
[IMP] Improve SMS module for V7. SMS and mass SMS work.

yvanpatry committed:
[ADD] Add two new module , add is_company in the domain of partner_id in the sale and the invoice

Commit:
add images

Commit:
add images

Commit:
add icon to smsclient module

Commit:
add icon to smsclient module

Commit:
add images for account import report

mathieuvatel committed:
[MRG] with hanane's images documentation

mathieuvatel committed:
[IMP] code cleaning on smsclient

Commit:
resolve conflicts

yvanpatry committed:
[IMP] Correct description

yvanpatry committed:
[IMP] Fix SOAPpy issue and URL know works

mathieuvatel committed:
[IMP] - import dependencies

mathieuvatel committed:
[IMP] - smsclient module's description

yvanpatry committed:
[ADD] Add contract customer invoice and migrate 
[IMP] SMS erase print
[IMP] Stock analytic XML part is now migrated

yvanpatry committed:
[MRG] Merge with pierre

yvanpatry committed:
[IMP] Add new porperty field for gateway

yvanpatry committed:
[ADD] Module account unrealized currency gain loss from v6.1

yvanpatry committed:
[IMP] Imp SMS_client options

yvanpatry committed:
[IMP] Fix

yvanpatry committed:
[FIX] Imp encodage

yvanpatry committed:
[IMP] Encodage issue

yvanpatry committed:
[FIX] Unicode

yvanpatry committed:
[FIX] SmsCLient Encode

yvanpatry committed:
[FIX]

yvanpatry committed:
[IMP] Encodage for the sms

yvanpatry committed:
[IMP] Optional Coding

mathieuvatel committed:
[IMP] remove prints

yvanpatry committed:
[IMP] SMS Client

Commit:
Initial commit

Commit:
Merge branch 'master' of github.com:akretion/smsclient

Commit:
move into subdirectory

Commit:
Portage from v7 to v8 with some reworking

Commit:
Portage wizard v7 to v8

Commit:
Commit before delete

Commit:
Delete old sms client

Commit:
New architecture for smsclient

Commit:
Updated openerp.py

Commit:
Add module to send sms when picking is ready to transfert

Commit:
Change according to selection

Commit:
Remove option in wizard

Commit:
Add depends

Commit:
Add only delivery filter, rename the module

Commit:
Exclude ropeproject files

Commit:
Merge remote-tracking branch 'origin/8.0' into 8.0-sms

yvaucher commented:
For contributor Hanane (no github login found) does he has a github account?

Viggor commented:
The module come from launchpad so he probably doesn't have a github account.

yvanpatry commented:
At my knowledge she don't have an github account and she no longer works with us so ! 

Cordialement, 
Yvan PATRY 

Julius Network Solutions 
32, rue du Général Foy 75008 Paris 
http://www.julius.fr/ 
Tel : +33 (0) 144 700 804 
Fax : +33 (0) 955 646 946 

De: "Viggor" notifications@github.com 
À: "OCA/connector-telephony" connector-telephony@noreply.github.com 
Cc: "yvan" yvan@julius.fr 
Envoyé: Mardi 4 Août 2015 17:25:53 
Objet: Re: [connector-telephony] 8.0 sms (#55) 

The module come from launchpad so he probably doesn't have a github account. 

— 
Reply to this email directly or view it on GitHub .

yvaucher commented on .gitignore:
```diff
@@ -47,6 +54,10 @@ coverage.xml
 
 # Rope
 .ropeproject
+*.pot
```
Don't add *.pot in gitignore it is used for translations

yvaucher commented on sms_send_picking/__openerp__.py:
```diff
@@ -0,0 +1,45 @@
+ 'website': 'www.akretion.com',
+ 'license': 'AGPL-3',
+ 'category': 'Generic Modules',
+ 'description': """
```
Description should go in a README.rst file and you have to remove it completely from `__openerp__.py`

yvaucher commented on sms_send_picking/__openerp__.py:
```diff
@@ -0,0 +1,45 @@
+ 'author': 'Akretion',
+ 'website': 'www.akretion.com',
+ 'license': 'AGPL-3',
+ 'category': 'Generic Modules',
```
Would `Phone` be better for category?

yvaucher commented on sms_send_picking/__openerp__.py:
```diff
@@ -0,0 +1,45 @@
+
+
+
+
```
Please remove useless empty lines at end of file

yvaucher commented on sms_send_picking/stock.py:
```diff
@@ -0,0 +1,55 @@
+    def _send_sms(self):
+        sms_sender_obj = self.env['partner.sms.send']
+        gateways = self.env['sms.smsclient'].search([('default_gateway', '=', True)])
+        gateway = gateways[0]
```
You can do a `self.env[...].search(domain, limit=1)` to ensure you have only one gateway.

yvaucher commented on sms_send_picking/stock.py:
```diff
@@ -0,0 +1,55 @@
+    def copy(self, default=None):
+        if default is None:
+            default = {}
+        default['sms_sent'] = False
```
Use `default` attribute when declaring the fields.Boolean instead.

yvaucher commented on smsclient_core/__init__.py:
```diff
@@ -0,0 +1,28 @@
+from . import wizard
+from . import smstemplate
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
We agreed to remove vim lines noone stated that he was using it.

yvaucher commented on sms_send_picking/__openerp__.py:
```diff
@@ -0,0 +1,45 @@
+
+{'name': 'Sms Send Picking',
+ 'version': '0.0.1',
+ 'author': 'Akretion',
```
Please add OCA in author

yvaucher commented on smsclient_core/__openerp__.py:
```diff
@@ -0,0 +1,60 @@
+                "email_template",
+                'base_phone',
+                ],
+    "author": "Julius Network Solutions,SYLEAM,OpenERP SA,Akretion",
```
Please add OCA in author

yvaucher commented on smsclient_core/__openerp__.py:
```diff
@@ -0,0 +1,60 @@
+                ],
+    "author": "Julius Network Solutions,SYLEAM,OpenERP SA,Akretion",
+    'images': ['images/sms.jpeg', 'images/gateway.jpeg', 'images/gateway_access.jpeg','images/client.jpeg','images/send_sms.jpeg'],
+    "description": """
```
description goes in README.rst

yvaucher replied:
And please provide credits for  smsclient_core/static/src/img/icon.png in README.rst

Here is a template to follow: https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

yvaucher commented on smsclient_core/smsclient_wizard.xml:
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
```
This file is not loaded. To remove?

yvaucher commented on smsclient_core/smsclient.py.backup:
```diff
@@ -0,0 +1,409 @@
+# -*- coding: utf-8 -*-
```
To cleanup

yvaucher commented on smsclient_core/serveraction_view.xml:
```diff
@@ -0,0 +1,27 @@
+                        </group>
+                    </page>
+                </page>
+                <!-- <field name="sms" position="attributes"> -->
```
Remove those commented lines

Commit:
Fix flake 8

yvaucher commented on smsclient_core/serveraction.py:
```diff
@@ -0,0 +1,157 @@
+    """
+    _inherit = 'ir.actions.server'
+
+    mobile = fields.Char('Mobile No', size=512, help="Provides fields that "
```
You can remove size or is it intended to limit to 512 ?

yvaucher commented on smsclient_core/serveraction.py:
```diff
@@ -0,0 +1,157 @@
+                        ('tag', '=', gateway.tag),
+                        ('nostop', '=', gateway.nostop)
+                        ])
+                    print sms_in_q
```
Is this intended for debugging? If so use logging. Or remove it.

yvaucher commented on smsclient_core/smsclient.py:
```diff
@@ -0,0 +1,440 @@
+                            default=_default_get_mobile)
+    app_id = fields.Char('API ID', size=256)
+    user = fields.Char('Login', size=256)
+    password = fields.Char('Password', size=256)
```
Size is not mandatory anymore. Any reason to define it for all those fields?

yvaucher commented on smsclient_core/smsclient.py:
```diff
@@ -0,0 +1,440 @@
+
+    name = fields.Char('Gateway Name', size=256, required=True)
+    url = fields.Char('Gateway URL', size=256,
+                      required=True, help='Base url for message')
```
size constraint to remove.

yvaucher commented on smsclient_core/smsclient.py:
```diff
@@ -0,0 +1,440 @@
+                                   'gateway_id', 'History')
+    method = fields.Selection(string='API Method',
+                              selection='get_method',
+                              select=True,
```
Please use `index` instead of `select`

yvaucher commented on smsclient_core/smsclient.py:
```diff
@@ -0,0 +1,440 @@
+        ('new', 'Not Verified'),
+        ('waiting', 'Waiting for Verification'),
+        ('confirm', 'Verified'),
+        ], 'Gateway Status', select=True, readonly=True, default='new')
```
s/select/index

yvaucher commented on smsclient_core/smsclient.py:
```diff
@@ -0,0 +1,440 @@
+    code = fields.Char('Verification Code', size=256)
+    code_visible = fields.Boolean(default=False)
+    body = fields.Text('Message',
+                       help="The message text that will be send along with the"
```
Warning your help message will concat "... along with theemail which ..."

yvaucher commented on smsclient_core/smsclient.py:
```diff
@@ -0,0 +1,440 @@
+                       help="The message text that will be send along with the"
+                            "email which is send through this server")
+    validity = fields.Integer('Validity',
+                              help='The maximum time - in minute(s) -'
```
Add a space before last `'`

yvaucher commented on smsclient_core/smsclient.py:
```diff
@@ -0,0 +1,440 @@
+        ('0', '0'),
+        ('1', '1'),
+        ('2', '2'),
+        ('3', '3')
```
I see this list multiple times. If it is the same values it could be a constant.

yvaucher commented on smsclient_core/smsclient.py:
```diff
@@ -0,0 +1,440 @@
+        ('0', 'Flash'),
+        ('1', 'Phone display'),
+        ('2', 'SIM'),
+        ('3', 'Toolkit')
```
Same here, put this selection in a constant

yvaucher commented on smsclient_core/smsclient.py:
```diff
@@ -0,0 +1,440 @@
+        default='1'
+    )
+    coding_visible = fields.Boolean(default=False)
+    tag = fields.Char('Tag', size=256, help='an optional tag')
```
Any reason for the `size` ?

yvaucher commented on smsclient_core/smsclient.py:
```diff
@@ -0,0 +1,440 @@
+
+    @api.onchange('method')
+    def onchange_method(self):
+        if self.method == '':
```
`if not self.method` ?

yvaucher commented on smsclient_core/smsclient.py:
```diff
@@ -0,0 +1,440 @@
+    @api.model
+    def create(self, vals):
+        super(HistoryLine, self).create(vals)
+        self.env.cr.commit()
```
You should certainly remove this commit and return the result of super. Or is there any reason to do it this way? If so a comment would be very appreciated but I think this is strange.

yvaucher commented on smsclient_core/smsclient_view.xml:
```diff
@@ -0,0 +1,194 @@
+                                    <field name="char_limit_visible" invisible="1"/>
+								</group>
+							</group>
+						</page>
```
Please fix the indentation of this file there are a bunch of tab we only want spaces

yvaucher commented on smsclient_core/smsclient_view.xml:
```diff
@@ -0,0 +1,194 @@
+                    <field name="gateway_id"/>
+                    <field name="to"/>
+                    <field name="sms"/>
+                </tree>
```
Maybe you want an editable view list here as form gives no advantages.

yvaucher commented on smsclient_core/__openerp__.py:
```diff
@@ -0,0 +1,60 @@
+    "description": """
+SMS Client module provides:
+-------------
+Sending SMSs very easily, individually or collectively.
```
This first line can definitively goes into `summary` of `__openerp__.py`

yvaucher commented on smsclient_core/__openerp__.py:
```diff
@@ -0,0 +1,60 @@
+updated to 8.0 by Akretion.
+    """,
+    "website": "http://julius.fr",
+    "category": "Tools",
```
Phone ?

yvaucher commented on smsclient_core/partner_sms_send_view.xml:
```diff
@@ -0,0 +1,47 @@
+            <field name="name">Send SMS</field>
+            <field name="model">partner.sms.send</field>
+            <field name="arch" type="xml">
+              <form string="SMS - Gateway: clickatell" version="7.0">
```
To remove `version="7.0"`

yvaucher commented on smsclient_core/partner_sms_send_view.xml:
```diff
@@ -0,0 +1,47 @@
+		                    <field name="app_id" invisible="1"/>
+	   				    </group>
+	   				    <group>
+							<field name="user" invisible="1"/>
```
Please fix that messy indentation

yvaucher commented on smsclient_core/smsclient.py:
```diff
@@ -0,0 +1,440 @@
+        for partner in partner_pool.browse(active_ids):
+            i += 1
+            res = partner.mobile
+        if i > 1:
```
Would be simple to check if len(active_ids) > 1

yvaucher commented on smsclient_core/smsclient.py:
```diff
@@ -0,0 +1,440 @@
+    @api.model
+    def _default_get_gateway(self):
+        sms_obj = self.env['sms.smsclient']
+        gateway_ids = sms_obj.search([], limit=1)
```
Please rename this variable you don't get ids but a single record set

yvaucher commented on smsclient_core/smsclient.py:
```diff
@@ -0,0 +1,440 @@
+    def _default_get_gateway(self):
+        sms_obj = self.env['sms.smsclient']
+        gateway_ids = sms_obj.search([], limit=1)
+        return gateway_ids and gateway_ids[0] or False
```
gateway_ids[0] == gateway_ids

yvaucher commented on smsclient_core/smsclient.py:
```diff
@@ -0,0 +1,440 @@
+        return gateway_ids and gateway_ids[0] or False
+
+    @api.multi
+    def onchange_gateway(self, gateway_id):
```
You might want to change this to new api to remove it from xml.

yvaucher commented on smsclient_core/smsclient.py:
```diff
@@ -0,0 +1,440 @@
+    classes_visible = fields.Boolean(default=False)
+    deferred = fields.Integer(
+        'Deferred',
+        help='The time -in minute(s)- to wait before sending the message',
```
s/time -in minute(s)/time - in minute(s)/

yvaucher commented on smsclient_ovh/__openerp__.py:
```diff
@@ -0,0 +1,48 @@
+                ],
+    "author": "Julius Network Solutions,SYLEAM,OpenERP SAi,Akretion",
+    'images': ['images/sms.jpeg', 'images/gateway.jpeg', 'images/gateway_access.jpeg','images/client.jpeg','images/send_sms.jpeg'],
+    "description": """
```
Another description to put in a README.rst

yvaucher commented on smsclient_ovh/__openerp__.py:
```diff
@@ -0,0 +1,48 @@
+                "email_template",
+                "smsclient_core",
+                ],
+    "author": "Julius Network Solutions,SYLEAM,OpenERP SAi,Akretion",
```
Missing OCA and type on OpenERP SA

yvaucher commented:
Thanks for the porting on github of those modules.
You have my inline comments as a code review.

Plus have a look at travis: https://travis-ci.org/OCA/connector-telephony/jobs/74086845

Commit:
Fixing multiple comments from @yvaucher

Viggor commented:
@yvaucher Thank you for this big review! I've fixed most of your comments.

Commit:
Fix flake8

Commit:
Put constant in uppercase and remove useless inherited method

Commit:
Remove unexpected characters

Commit:
Fix wrong logger syntax

Commit:
Add README.rst files in sms modules

Commit:
Add correct OCA author name

Commit:
Add a space for proper concat

Viggor commented:
Author name correctly added and space added for proper concat. Thanks for all!

yvaucher commented:
@Viggor You might have missed a comment on smsclient_core/smsclient.py on priority field

Commit:
Missed a priority field to put constant list

Viggor commented:
The travis errors come from the module base_phone, so everything is ok on my side.

Commit:
Ad translation

sebastienbeau commented on smsclient_core/smsclient.py:
```diff
@@ -0,0 +1,401 @@
+             'this is not an advertising message')
+
+
+class Properties(models.Model):
```
It seem that this class is no longer used. Remove it if it's dead code

sebastienbeau commented on smsclient_core/smsclient.py:
```diff
@@ -0,0 +1,401 @@
+        help='If parameter concern a value to substitute, indicate it')
+
+
+class HistoryLine(models.Model):
```
No sure that an HistoryLine object is needed. We can just keep the mail send in the queue with the state send for the history, what do you think?

sebastienbeau commented on smsclient_core/wizard/sendcode.py:
```diff
@@ -0,0 +1,62 @@
+                                    'code': msg})
+        return {}
+
+#    states = {
```
if it's dead code, remove it

sebastienbeau commented on smsclient_core/wizard/verifycode.py:
```diff
@@ -0,0 +1,78 @@
+                'Verification failed. Invalid Verification Code!'))
+        return {}
+
+    states = {
```
I think this code come from V5, not needed anymore on 8

sebastienbeau commented on smsclient_core/smsclient.py:
```diff
@@ -0,0 +1,401 @@
+class SMSQueue(models.Model):
+    _name = 'sms.smsclient.queue'
+    _description = 'SMS Queue'
+
```
Add teh company id so we can only see the sms of our company

Commit:
Add server environnement support

Commit:
Fix server environment support

sebastienbeau committed:
[REF][DEPRECATED] remove deprecated code for an old time

sebastienbeau committed:
[REF] start to refactor remove model history line as we can simple have a state on the sms send

sebastienbeau committed:
[REF] refactor code according to new api

sebastienbeau committed:
[REF] rename sms.client.queue into sms.sms

sebastienbeau committed:
[REF] start to refactor the API

sebastienbeau committed:
[REF] continue refactoring, improve api, add boutton on sms view for sending/retrying/cancelling, improve and fix wizard for sending massive sms

sebastienbeau committed:
[IMP] add a basic error management for OVH case

sebastienbeau committed:
[FIX] fix partner_id on wrong model and improve sms view to show partner and mobile number

sebastienbeau committed:
[REF] remove wizard for sending an sms, indeed we can send sms using the mass wizard, this will avoid duplicated code

sebastienbeau committed:
[REF] clean view and reorder menu

sebastienbeau committed:
[REF] rename sms.smsclient to sms.gateway and use the word gateway everywhere

sebastienbeau committed:
[REF] reafactor getting default gateway, and fix ir.model.access

sebastienbeau committed:
[REF] remove tag as it's not needed, maybe we may reimplement it latter with a m2m_tag

sebastienbeau committed:
[REF] comment code as it's broken for now

sebastienbeau committed:
[REF] refactor sending sms for picking, add a todo for using template

sebastienbeau committed:
[REF] continue refactoring, fix some error, add multi-company support

sebastienbeau committed:
[REVERT] revert commit 5f0cb5f444950cce28eafec2c07e78cd25467639, indead tag is need

sebastienbeau committed:
[FIX] fix encoding issue by using the beautiful request

sebastienbeau committed:
[FIX] remove pdb and fix access right

sebastienbeau committed:
[REF] rename method in order to be more homogenous

legalsylvain commented on .gitignore:
```diff
@@ -12,6 +12,7 @@ bin/
 build/
 develop-eggs/
 dist/
+downloads/
```
please remove changes in gitignore.

legalsylvain commented on smsclient_ovh/sms_gateway.py:
```diff
@@ -0,0 +1,91 @@
+        params = self._prepare_http_ovh().items()
+        r = requests.get(self.gateway_id.url, params=params)
+        _logger.debug("Call OVH API : %s params %s",
+                      self.gateway_id.url, params)
```
I know that this is a debug logger but is it great to have gateway password in log ?

sebastienbeau replied:
Good catch ! thanks !

sebastienbeau committed:
[FIX] fix wrong method was called

ahilali committed:
[IMP] make all fields readonly when message was sent

Commit:
Add period at the end of some help sentence, clean dead code, reformat
header and versionning according to OCA.

sebastienbeau committed:
Revert "[IMP] make all fields readonly when message was sent"

This reverts commit 274dc7f39eecf771b543f27cc10a826525d5368a.

sebastienbeau committed:
[IMP] make field in readonly

sebastienbeau committed:
[FIX] hid password in log

bealdav committed:
[FIX] sent instead of send and cosmetic typo

bealdav committed:
Merge pull request #2 from akretion/8-sms-sent-fix

[FIX] sent instead of send and cosmetic typo

sebastienbeau committed:
[IMP] add better logging and commit after generating the sms

sebastienbeau committed:
[FIX] fix hidding password

sebastienbeau committed:
[IMP] hide more data in debug mode

yvaucher commented on sms_send_picking/README.rst:
```diff
@@ -0,0 +1,64 @@
+    :alt: License: AGPL-3
+
+SMS send picking
+===========
```
Please update this README according to last version:

https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst

Title must have enough `=`

yvaucher commented on sms_send_picking/README.rst:
```diff
@@ -0,0 +1,64 @@
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/{project_repo}/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/{project_repo}/issues/new?body=module:%20{module_name}%0Aversion:%20{version}%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Update this part by removing this link

yvaucher commented on sms_send_picking/README.rst:
```diff
@@ -0,0 +1,64 @@
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/{project_repo}/issues>`_.
```
replace `{project_repo}` place holder

yvaucher commented on sms_send_picking/README.rst:
```diff
@@ -0,0 +1,64 @@
+
+For further information, please visit:
+
+* https://www.odoo.com/forum/help-1
```
Remove this link and add the Try me on runbot button

yvaucher commented on sms_send_picking/stock.py:
```diff
@@ -0,0 +1,73 @@
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
```
Use short header

yvaucher commented on sms_send_picking/__openerp__.py:
```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+# © 2015 Valentin CHEMIERE <valentin.chemiere@akretion.com>
```
Prefer Copyright instead of `©`

yvaucher commented on sms_send_picking/stock.py:
```diff
@@ -0,0 +1,73 @@
+    availability_sent_by_sms = fields.Boolean(default=False)
+
+    #TODO use a templating instead
+    @api.model
```
In this method you are using `self.partner_id` and `self.company_id` thus it should be `api.multi`

yvaucher commented on sms_send_picking/stock.py:
```diff
@@ -0,0 +1,73 @@
+            _logger.debug('Send Sms for picking %s, progress %s/%s', picking, idx, total)
+            vals = picking._prepare_availability_by_sms_notification()
+            self.env['sms.sms'].create(vals)
+            picking.write({'availability_sent_by_sms': True})
```
As you write a single field you can do:

``` python
picking.availability_sent_by_sms = True
```

yvaucher commented on smsclient_core/README.rst:
```diff
@@ -0,0 +1,69 @@
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/{project_repo}/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/{project_repo}/issues/new?body=module:%20{module_name}%0Aversion:%20{version}%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Update this file according to https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst

yvaucher commented on smsclient_core/README.rst:
```diff
@@ -0,0 +1,69 @@
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/{project_repo}/issues>`_.
```
Replace `{project_repo}`

yvaucher commented on smsclient_core/README.rst:
```diff
@@ -0,0 +1,69 @@
+* select on or multiple partner
+* go to more and will found some wizards to send sms
+
+
```
Add "Try me on runbot" button

yvaucher commented on smsclient_core/__init__.py:
```diff
@@ -0,0 +1,26 @@
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
```
You can remove the license header from `__init__.py` file or you must use the small license header:
https://github.com/OCA/maintainer-tools/blob/master/template/module/models/model_name.py#L1-L3

yvaucher commented on smsclient_core/__openerp__.py:
```diff
@@ -0,0 +1,43 @@
+# © 2004-2009 OpenERP SA (<http://openerp.com>)
+# © 2011 SYLEAM (<http://syleam.fr/>)
+# © 2013 Julius Network Solutions SARL <contact@julius.fr>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
```
Update this to the last version of the small license header:

https://github.com/OCA/maintainer-tools/blob/master/template/module/models/model_name.py#L1-L3

yvaucher commented on smsclient_core/serveraction.py:
```diff
@@ -0,0 +1,32 @@
+# Copyright (C) 2004-2009 Tiny SPRL (<http://tiny.be>).
+# Copyright (C) 2013 Julius Network Solutions SARL <contact@julius.fr>
+# Copyright (C) 2015 Valentin Chemiere <valentin.chemiere@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
```
https://github.com/OCA/maintainer-tools/blob/master/template/module/models/model_name.py#L1-L3

yvaucher commented on smsclient_core/serveraction.py:
```diff
@@ -0,0 +1,32 @@
+        help="Field to be used to fetch the mobile number, e.g. you select"
+             " the invoice model and `object.invoice_address_id.mobile` "
+             "will be the field providing the correct mobile number.")
+    sms = fields.Char('SMS', size=160, translate=True)
```
Don't we need a constraint on size ? 160 char is for 7-bit characters (ASCII). You won't be prevented to enter non ASCII characters here.

yvaucher commented on smsclient_core/sms_gateway.py:
```diff
@@ -0,0 +1,266 @@
+# Copyright (C) 2013 Julius Network Solutions SARL <contact@julius.fr>
+# Copyright (C) 2015 Valentin Chemiere <valentin.chemiere@akretion.com>
+# Copyright (C) 2015 Sébastien BEAU <sebastien.beau@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
```
https://github.com/OCA/maintainer-tools/blob/master/template/module/models/model_name.py#L1-L3

yvaucher commented on smsclient_core/sms_gateway.py:
```diff
@@ -0,0 +1,266 @@
+    @api.multi
+    def send(self):
+        for sms in self:
+            if sms.gateway_id.char_limit and len(sms.message) > 160:
```
This doesn't handle if `coding` is unicode plus message is very likely to include non-ascii characters.

It would be a good idea to make a function for this check as you might want to use it at different places.

yvaucher commented on smsclient_core/sms_gateway_view.xml:
```diff
@@ -0,0 +1,165 @@
+            <field name="arch" type="xml">
+                <form string="SMS Gateway">
+                    <sheet>
+                        <group col="2">
```
What you probably want here is a

``` xml
<div class="oe_title">
    <h1><field name="name"/></h1>
    <field name="Method" placeholder="Method"/>
</div>
```

yvaucher commented on smsclient_core/sms_gateway_view.xml:
```diff
@@ -0,0 +1,165 @@
+                        <field name="state" widget="statusbar" statusbar_visible="draft,send"/>
+                    </header>
+                    <sheet>
+                        <label for="partner_id" class="oe_edit_only"/>
```
Put this in a `<div class="oe_title">`

yvaucher commented on smsclient_core/sms_gateway_view.xml:
```diff
@@ -0,0 +1,165 @@
+        <record model="ir.ui.view" id="sms_sms_tree">
+            <field name="model">sms.sms</field>
+            <field name="arch" type="xml">
+                <tree string="Sms">
```
What about adding `colors="red:state == 'error'; blue: state == 'draft'; grey: state == 'send'"`

yvaucher commented on smsclient_core/smstemplate.py:
```diff
@@ -0,0 +1,16 @@
+# Copyright (C) 2004-2009 Tiny SPRL (<http://tiny.be>).
+# Copyright (C) 2013 Julius Network Solutions SARL <contact@julius.fr>
+# Copyright (C) 2015 Valentin Chemiere <valentin.chemiere@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
```
https://github.com/OCA/maintainer-tools/blob/master/template/module/models/model_name.py#L1-L3

yvaucher commented on smsclient_core/smstemplate_view.xml:
```diff
@@ -0,0 +1,23 @@
+				<field name="lang" position="before">
+					<field name="sms_template"/>
+					<field name="gateway_id"/>
+				</field>
```
Please don't mix tab and spaces

yvaucher commented on smsclient_core/wizard/__init__.py:
```diff
@@ -0,0 +1,22 @@
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>
+#
+##############################################################################
```
No header or short header

yvaucher commented on smsclient_core/wizard/mass_sms.py:
```diff
@@ -0,0 +1,97 @@
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>
+#
+##############################################################################
```
Short header

yvaucher commented on smsclient_core/wizard/mass_sms.py:
```diff
@@ -0,0 +1,97 @@
+                    'coding', 'tag', 'nostop']:
+            self[key] = self.gateway_id[key]
+
+    @api.model
```
`@api.multi`

yvaucher commented on smsclient_core/wizard/mass_sms_view.xml:
```diff
@@ -0,0 +1,55 @@
+        <record model="ir.ui.view" id="view_wizard_send_sms">
+            <field name="model">wizard.send.sms</field>
+            <field name="arch" type="xml">
+            	<form string="SMS Gateway - Sens SMS" version="7.0">
```
Remove `version="7.0"`

yvaucher commented on smsclient_core/wizard/mass_sms_view.xml:
```diff
@@ -0,0 +1,55 @@
+	                <group string="Gateway" colspan="4">
+                        <field name="gateway_id"/>
+					</group>
+					<notebook>
```
Don't mix tab and spaces

yvaucher commented on smsclient_core/wizard/mass_sms_view.xml:
```diff
@@ -0,0 +1,55 @@
+                            <field name="message" colspan="4" nolabel="1"/>
+                            <!-- TODO limit to 10 the number of partner to show -->
+                            <field name="partner_ids" colspan="4" nolabel="1">
+                                <tree options="{'limit': 10}">
```
My wild guess is that `options` is not working, please remove it.

yvaucher commented on smsclient_core/wizard/mass_sms_view.xml:
```diff
@@ -0,0 +1,55 @@
+                                </tree>
+                            </field>
+                        </page>
+	                   	<page string="Additional option">
```
s/option/options

yvaucher commented on smsclient_ovh/README.rst:
```diff
@@ -0,0 +1,57 @@
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/{project_repo}/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/{project_repo}/issues/new?body=module:%20{module_name}%0Aversion:%20{version}%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Readme file to update

yvaucher commented on smsclient_ovh/__init__.py:
```diff
@@ -0,0 +1,23 @@
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
```
No header or short header

yvaucher commented on smsclient_ovh/__openerp__.py:
```diff
@@ -0,0 +1,49 @@
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
```
Short header

yvaucher commented on smsclient_ovh/__openerp__.py:
```diff
@@ -0,0 +1,49 @@
+    "installable": True,
+}
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
You can remove the vim line

yvaucher commented on smsclient_ovh/sms_gateway.py:
```diff
@@ -0,0 +1,97 @@
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
```
Short header

yvaucher commented on smsclient_ovh/__openerp__.py:
```diff
@@ -0,0 +1,49 @@
+
+{
+    "name": "SMS Client OVH",
+    "version": "1.0",
```
`8.0.1.0.0`

yvaucher commented on smsclient_ovh/sms_gateway.py:
```diff
@@ -0,0 +1,97 @@
+class SmsSms(models.Model):
+    _inherit = "sms.sms"
+
+    @api.model
```
`@api.multi`

Commit:
Add module smsclient_smsmode

smsclient_core: SMS should not be readable by everybody by default: delete corresponding ir.model.access
wizard to mass send SMS should be restricted to members of the SMS User group
Update and simplify default ir.rule

Commit:
smsclient_core: improve ir.model.access

Commit:
smsclient_core: minor code improvements

alexis-via committed:
smsclient_core: FR translation improvements

alexis-via committed:
smsclient_core: Improve mass SMS wizard code

This change make it easier to run the wizard from employees or
applicants or any other object linked to partners.
